examples/tensorrt_llm/common/base_engine.py

```diff
@@ -15,88 +15,123 @@
+    cleaned_dict = {
+        key: value for key, value in sampling_params.items() if not key.startswith("_")
+    }
+    return SamplingParams(**cleaned_dict)
 
 
-class BaseTensorrtLLMEngine(ChatProcessorMixin):
+class BaseTensorrtLLMEngine:
     def __init__(
         self,
-        trt_llm_engine_config: TensorrtLLMEngineConfig,
+        namespace_str: str = "dynamo",
+        component_str: str = "tensorrt-llm",
+        worker_id: Optional[str] = None,
+        engine_config: LLMAPIConfig = None,
+        remote_prefill: bool = False,
+        min_workers: int = 0,
+        disagg_config_file: Optional[str] = None,
+        block_size: int = 32,
+        router: str = "round_robin",
+        server_type: ServerType = ServerType.GEN,
     ):
-        super().__init__(trt_llm_engine_config.engine_config)
-        self._namespace_str = trt_llm_engine_config.namespace_str
-        self._component_str = trt_llm_engine_config.component_str
-        self._worker_id = trt_llm_engine_config.worker_id
-        self._kv_metrics_publisher = trt_llm_engine_config.kv_metrics_publisher
-        self._publish_stats = trt_llm_engine_config.publish_stats
-        self._publish_kv_cache_events = trt_llm_engine_config.publish_kv_cache_events
-        self._kv_block_size = trt_llm_engine_config.kv_block_size
-        self._error_queue: Optional[Queue] = None
-
-        self._init_engine()
+        self._namespace_str = namespace_str
+        self._component_str = component_str
+        self._worker_id = worker_id
+        self._remote_prefill = remote_prefill
+        self._min_workers = 0
+        self._kv_block_size = block_size
+        self._router = router
+        self._server_type = server_type
+        self._prefill_client = None
+        self._error_queue = None
+        self._kv_metrics_publisher = None
+
+        if self._remote_prefill:
+            self._min_workers = min_workers
+            if disagg_config_file is None or not os.path.exists(disagg_config_file):
+                raise ValueError(
+                    "llmapi_disaggregated_config file does not exist or not provided"
+                )
+            disagg_config = parse_disagg_config_file(disagg_config_file)
+            server_config: CtxGenServerConfig = None
+
+            for config in disagg_config.server_configs:
+                # Select the first context server config
+                if config.type == server_type.value:
+                    server_config = config
+                    break
+
+            if server_config is None:
+                server_type_str = (
+                    "generation" if server_type == ServerType.GEN else "context"
+                )
+                raise ValueError(
+                    f"No {server_type_str} server config found. Please check the disaggregated config file."
+                )
+
+            engine_config = update_args_from_disagg_config(engine_config, server_config)
+
+        if router == "kv":
```

Comment:
nit: would move up to the top and initialize to false
```
self._publish_stats = False
self._publish_events = False
if router == "kv":
   self._publish_stats = True
   self._publish_events = True
   self._kv_metrics_publisher = KvMetricsPublisher()
```

you may be able to get rid of publish stats and treat `self._kv_metrics_publisher` as a None or filled in
```

Reply:
Will resolve it on a later PR.